### ISSUE TITLE
new clang warning about expressions potentially be evaluated when used in typeid() "-Wpotentially-evaluated-expression" (Trac #1017)

### ISSUE THREAD

**Issue by icecube-trac** [icecube-trac/tix2#1011]
new versions of clang seem to have learned about more things to warn about: 

```text
tableio/public/tableio/I3Converter.h:278:40: warning: expression with side effects will be evaluated despite
      being used as an operand to 'typeid' [-Wpotentially-evaluated-expression]
      return (typeid(FrmObj) == typeid(*object.get()));
                                       ^

```

worth a ticket?

Migrated from https://code.icecube.wisc.edu/ticket/1017
```json
{
    "status": "closed", 
    "changetime": "2019-02-13T14:11:42", 
    "description": "new versions of clang seem to have learned about more things to warn about: \n{{{\ntableio/public/tableio/I3Converter.h:278:40: warning: expression with side effects will be evaluated despite\n      being used as an operand to 'typeid' [-Wpotentially-evaluated-expression]\n      return (typeid(FrmObj) == typeid(*object.get()));\n                                       ^\n\n}}}\n\nworth a ticket?", 
    "reporter": "claudio.kopper", 
    "cc": "", 
    "resolution": "fixed", 
    "_ts": "1550067102580394", 
    "component": "combo core", 
    "summary": "new clang warning about expressions potentially be evaluated when used in typeid() \"-Wpotentially-evaluated-expression\"", 
    "priority": "minor", 
    "keywords": "", 
    "time": "2015-06-11T01:57:40", 
    "milestone": "", 
    "owner": "nega", 
    "type": "cleanup"
}
```

**Comment by icecube-trac** [comment]
Trac update at 20150810T20:50:46: olivas changed milestone from "" to "January 2016"

**Comment by icecube-trac** [comment]
Trac update at 20150810T23:54:28: 
* olivas changed owner from "" to "olivas, nega"
* olivas changed status from "new" to "assigned"

**Comment by icecube-trac** [comment]
Trac update at 20150811T02:19:47: nega changed owner from "olivas, nega" to "claudio.kopper"

**Comment by icecube-trac** [comment]
Trac update at 20160216T23:05:39: 
* claudio.kopper changed priority from "normal" to "minor"
* claudio.kopper commented:

> re-assign this to Don, lower the priority (this is really, really not important), and bump the milestone.
> 
> A way to fix this would be to split this up into two steps - de-reference *object.get() first, assign it to a temporary variable and then use typeid on that.


* claudio.kopper changed milestone from "February 2016" to "April 2016"
* claudio.kopper changed owner from "claudio.kopper" to "nega"

**Comment by icecube-trac** [comment]
Trac update at 20160217T02:32:50: 
* nega changed resolution from "" to "worksforme"
* nega changed status from "assigned" to "closed"
* nega commented:

> can't reproduce with clang 3.8RC and our default warnings

**Comment by icecube-trac** [comment]
Trac update at 20160217T03:08:27: 
* nega commented:

> occurs in only Debug builds


* nega changed status from "closed" to "reopened"
* nega changed resolution from "worksforme" to ""

**Comment by icecube-trac** [comment]
Trac update at 20160330T19:54:55: 
* nega commented:

> In changeset 2499:
> 
> ```text
> #!CommitTicketReference repository="IceTray" revision="2499"
> set a warning suppression flag
> 
> closes #1011
> ```


* nega changed status from "reopened" to "closed"
* nega changed resolution from "" to "fixed"

**Comment by icecube-trac** [comment]
Trac update at 20190112T00:13:29: nwhitehorn commented:

> In changeset 170230:
> 
> ```text
> #!CommitTicketReference repository="IceCube" revision="170230"
> set a warning suppression flag
> 
> closes #1011
> ```

**Comment by icecube-trac** [comment]
Trac update at 20190213T14:11:42: 
* olivas commented:

> Milestone April 2016 deleted


* olivas changed milestone from "April 2016" to ""
